lock/src/test/java/org/infinispan/lock/ClusteredLockSplitBrainTest.java

```diff
@@ -59,118 +79,132 @@ public void testLockUseAfterPartitionWithoutMajority() throws Throwable {
       ClusteredLockManager clm1 = EmbeddedClusteredLockManagerFactory.from(getCacheManagers().get(1));
       ClusteredLockManager clm2 = EmbeddedClusteredLockManagerFactory.from(getCacheManagers().get(2));
       ClusteredLockManager clm3 = EmbeddedClusteredLockManagerFactory.from(getCacheManagers().get(3));
+      ClusteredLockManager clm4 = EmbeddedClusteredLockManagerFactory.from(getCacheManagers().get(4));
+      ClusteredLockManager clm5 = EmbeddedClusteredLockManagerFactory.from(getCacheManagers().get(5));
 
       clm0.defineLock(LOCK_NAME);
-
       assertTrue(clm0.isDefined(LOCK_NAME));
-      ClusteredLock lock0 = clm0.get(LOCK_NAME);
-      assertNotNull(lock0);
 
+      ClusteredLock lock0 = clm0.get(LOCK_NAME);
       ClusteredLock lock1 = clm1.get(LOCK_NAME);
-      assertNotNull(lock1);
-
       ClusteredLock lock2 = clm2.get(LOCK_NAME);
-      assertNotNull(lock2);
-
       ClusteredLock lock3 = clm3.get(LOCK_NAME);
-      assertNotNull(lock3);
-
-      splitCluster(new int[]{0, 1}, new int[]{2, 3});
-
-      assertNull(await(lock0.tryLock(5, TimeUnit.SECONDS).exceptionally(ex -> {
-         assertException(ClusteredLockException.class, ex);
-         assertException(AvailabilityException.class, ex.getCause());
-         return null;
-      })));
-      assertNull(await(lock1.tryLock(5, TimeUnit.SECONDS).exceptionally(ex -> {
-         assertException(ClusteredLockException.class, ex);
-         assertException(AvailabilityException.class, ex.getCause());
-         return null;
-      })));
-      assertNull(await(lock2.tryLock(5, TimeUnit.SECONDS).exceptionally(ex -> {
-         assertException(ClusteredLockException.class, ex);
-         assertException(AvailabilityException.class, ex.getCause());
-         return null;
-      })));
-      assertNull(await(lock3.tryLock(5, TimeUnit.SECONDS).exceptionally(ex -> {
-         assertException(ClusteredLockException.class, ex);
-         assertException(AvailabilityException.class, ex.getCause());
-         return null;
-      })));
+      ClusteredLock lock4 = clm4.get(LOCK_NAME);
+      ClusteredLock lock5 = clm5.get(LOCK_NAME);
+
+      splitCluster(new int[]{0, 1, 2}, new int[]{3, 4, 5});
+
+      asList(lock0, lock1, lock2, lock3, lock4, lock5).forEach(lock -> {
+         assertNotNull(lock);
+         assertFalse(await(lock.tryLock(100, TimeUnit.MILLISECONDS).exceptionally(ex -> {
+            assertException(ClusteredLockException.class, ex);
+            assertException(AvailabilityException.class, ex.getCause());
```

Comment:
Could use `assertException(ClusteredLockException.class, AvailabilityException.class, ex)`, or even better`expectExecutionException(ClusteredLockException.class, AvailabilityException.class, lock.tryLock(100, MILLISECONDS)` if you expect the call to fail.

Reply:
ok